modules/space/rendering/renderabletravelspeed.cpp

```diff
@@ -0,0 +1,354 @@
+
+    ghoul::opengl::updateUniformLocations(*_shaderProgram, _uniformCache, UniformNames);
+
+} // initializeGL
+
+void RenderableTravelSpeed::deinitializeGL() {
+    BaseModule::ProgramObjectManager.release(
+        "Travelspeed",
+        [](ghoul::opengl::ProgramObject* p) {
+            global::renderEngine->removeRenderProgram(p);
+        }
+    );
+    glDeleteVertexArrays(1, &_vaoId);
+    glDeleteBuffers(1, &_vBufferId);
+} // deinitializeGL
+
+double RenderableTravelSpeed::calculateLightTravelTime(glm::dvec3 startPosition,
+    glm::dvec3 targetPosition) {
+    return glm::distance(targetPosition, startPosition) /
+        _travelSpeed;
+}
+
+void RenderableTravelSpeed::calculateDirectionVector() {
+    _directionVector = glm::normalize(_targetPosition - _sourcePosition);
+}
+
+void RenderableTravelSpeed::calculateVerticesPositions() {
+
+    // 3: start of light, 2: start of fade, 1: end of fade
+    _verticesPositions[2].x = _travelSpeed * _timeSinceStart * _directionVector.x;
+    _verticesPositions[2].y = _travelSpeed * _timeSinceStart * _directionVector.y;
+    _verticesPositions[2].z = _travelSpeed * _timeSinceStart * _directionVector.z;
+
+    // This if statment is there to not start the line from behind the source node
+    if (_timeSinceStart < _indicatorLength) {
+        _verticesPositions[1] = glm::vec3(0.0, 0.0, 0.0); // = source node
+    }
+    else {
+        _verticesPositions[1].x =
+            _travelSpeed * _directionVector.x * (_timeSinceStart - _indicatorLength);        
+        _verticesPositions[1].y =
+            _travelSpeed * _directionVector.y * (_timeSinceStart - _indicatorLength);
+        _verticesPositions[1].z =
+            _travelSpeed * _directionVector.z * (_timeSinceStart - _indicatorLength);
+
+    }
+
+    // This if statment is there to not start the line from behind the source node
+    if (_timeSinceStart < (_indicatorLength + _fadeLength)) {
+        _verticesPositions[0] = glm::vec3(0.0, 0.0, 0.0); // = source node
+    }
+    else {
+        _verticesPositions[0].x =
+            _travelSpeed * _directionVector.x * 
+            (_timeSinceStart - _indicatorLength - _fadeLength);
+        _verticesPositions[0].y =
+            _travelSpeed * _directionVector.y * 
+            (_timeSinceStart - _indicatorLength - _fadeLength);
+        _verticesPositions[0].z =
+            _travelSpeed * _directionVector.z * 
+            (_timeSinceStart - _indicatorLength - _fadeLength);
+    }
+}
+
+void RenderableTravelSpeed::updateVertexData() {
+
+    calculateVerticesPositions();
+
+    glBindVertexArray(_vaoId);
+    glBindBuffer(GL_ARRAY_BUFFER, _vBufferId);
+    glBufferData(
+        GL_ARRAY_BUFFER,
+        sizeof(_verticesPositions),
+        _verticesPositions,
+        GL_DYNAMIC_DRAW
+    );
+    glEnableVertexAttribArray(0);
+    glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 3*sizeof(float), nullptr);
+    glBindVertexArray(0);
+}
+
+void RenderableTravelSpeed::reinitiateTravel() {
```

Comment:
You should pass the time into this function rather than accessing the timeManager

Reply:
I'm doing it because I don't have access to the data.time.j2000Seconds() in the onChange functions declared in the constructor. At least I don't think so?
I'm open to change it some other way.